android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerOrchestrator.kt

```diff
@@ -182,21 +159,22 @@ class GestureHandlerOrchestrator(
       shouldResetProgress = true
       activationIndex = this@GestureHandlerOrchestrator.activationIndex++
     }
-    var toCancelCount = 0
+
     // Cancel all handlers that are required to be cancel upon current handler's activation
-    for (i in 0 until gestureHandlersCount) {
-      val otherHandler = gestureHandlers[i]!!
+    for (otherHandler in gestureHandlers) {
       if (shouldHandlerBeCancelledBy(otherHandler, handler)) {
-        handlersToCancel[toCancelCount++] = otherHandler
+        handlersToCancel.add(otherHandler)
       }
     }
-    for (i in toCancelCount - 1 downTo 0) {
-      handlersToCancel[i]!!.cancel()
+
+    for (otherHandler in handlersToCancel.reversed()) {
+      otherHandler.cancel()
     }
 
+    handlersToCancel.clear()
```

Comment:
Do we need `handlersToCancel` at all here? I think you can iterate over `gestureHandlers.asReversed()` and just cancel relevant handlers inside the loop.

Reply:
I agree - done in [da2c421](https://github.com/software-mansion/react-native-gesture-handler/pull/2672/commits/da2c4211da791d8b347723d90e77a0a7a4db29b6)